CHANGELOG.md

```diff
@@ -3,6 +3,233 @@
+  this is consistent with the behavior of project commands (e.g., `uv sync`), it's surprising to users
+  that are using imperative workflows (i.e., `uv pip`). Now, `uv venv` will prompt for confirmation
+  before removing an existing virtual environment. **If not in an interactive context, uv will still
+  remove the virtual environment for backwards compatibility. However, this behavior is likely to
+  change in a future release.**
+
+  The behavior for other commands (e.g., `uv sync`) is unchanged.
+
+  You can opt out of this behavior by setting `UV_VENV_CLEAR=1` or passing the `--clear` flag.
+
+- **Validate that discovered interpreters meet the Python preference ([#7934](https://github.com/astral-sh/uv/pull/7934))**
+
+  uv allows opting out of its managed Python versions with the `--no-managed-python` and
+  `python-preference` options.
+
+  Previously, uv would not enforce this option for Python interpreters discovered on the `PATH`. For
+  example, if a symlink to a managed Python interpreter was created, uv would allow it to be used
+  even if `--no-managed-python` was provided. Now, uv ignores Python interpreters that do not match
+  the Python preference _unless_ they are in an active virtual environment or are explicitly
+  requested, e.g., with `--python /path/to/python3.13`.
+
+  Similarly, uv would previously not invalidate existing project environments if they did not match
+  the Python preference. Now, uv will invalidate and recreate project environments when the Python
+  preference changes.
+
+  You can opt out of this behavior by providing the explicit path to the Python interpreter
+  providing `--managed-python` / `--no-managed-python` matching the interpreter you want.
+
+- **Install dependencies without build systems when they are `path` sources ([#14413](https://github.com/astral-sh/uv/pull/14413))**
+
+  When working on a project, uv uses the [presence of a build
+  system](https://docs.astral.sh/uv/concepts/projects/config/#build-systems) to determine if it
+  should be built and installed into the environment. However, when a project is a dependency of
+  another project, it can be surprising for the dependency to be missing from the environment.
+
+  Previously, uv would not build and install dependencies with [`path`
+  sources](https://docs.astral.sh/uv/concepts/projects/dependencies/#path) unless they declared a
+  build system or set `tool.uv.package = true`. Now, dependencies with `path` sources are built and
+  installed regardless of the presence of a build system. If a build system is not present, the
+  `setuptools.build_meta:__legacy__ ` backend will be used (per the standards).
+
+  You can opt out of this behavior by setting `package = false` in the source declaration, e.g.:
+
+  ```toml
+  [tool.uv.sources]
+  foo = { path = "./foo", package = false }
+  ```
+
+  Or, by setting `tool.uv.package = false` in the dependent `pyproject.toml`.
+
+  See the documentation on [virtual
```

Comment:
To avoid alarming users about checking what their glibc version is, I'd add a sentence that this only affects user of that specific option (which is rare), we use the host glibc version for all other use cases such as `uv pip install` and `uv sync`.

Reply:
Yes, that's my concern - it took me a while to understand that this does not affect normal resolution _or_ normal builds (since uv doesn't do builds in manylinux containers, that's cibuildwheel's job).